next -> redux opinionated

sergiokopplin opened the pull request:
Might need some help here.

ctrlplusb committed:
Adds modernizr with illustrative example. closes #256

ctrlplusb committed:
Linting now occurs on config, src and tools. closes #314

ctrlplusb committed:
Updates dependencies.

markyfyi committed:
update broken feature/flow link in docs

codepunkt committed:
Merge pull request #324 from markshlick/patch-1

update broken feature/flow link in docs

ctrlplusb committed:
Merge branch 'master' into next

ctrlplusb committed:
Updates dependencies.

rlindskog committed:
Updated CSP to add backwards compatibility to nonce. (#344)

* Updated CSP for nonce backwards compatibility

* Documented

* Updates comments

ctrlplusb committed:
- Renamed environment variables:
   - `SERVER_PORT` to `PORT`
   - `SERVER_HOST` to `HOST`
   - `CLIENT_DEVSERVER_PORT` to `CLIENT_DEV_PORT`
 - Replaces `code-split-component` with `react-async-component`
 - Renames the `nodeBundlesIncludeNodeModuleFileTypes` config property to `nodeExternalsFileTypeWhitelist`
 - Refactors the server and serviceworker offline page generation. We now use a set of React components (`ServerHTML` and `HTML`) to manage our HTML in a uniform fashion.
 - Refactors how we resolve environment specific configuration values. `NODE_ENV` is reserved for specifying a `development` or `production` build now. Use `CONF_ENV` to specify a target environment if you would like to resolve an environment specific .env file.
 - Refactors the client configuration filter rule to be contained within the main configuration and moves the configuration object creation into the server middleware.
 - Renames the `safeGetConfig` to `config`, and made it a default export.
 - Removes the `cross-env` library.

 - All server/client/shared code all use the shared config helper.
 - Updated dependencies, including to the latest Webpack official 2 release.

 - New babel plugins to optimise React production build performance.
 - Adds new icon sets.

 - Chrome favicon request issue.
 - Cleans up the package scripts.
 - Service worker would fail if a subfolder was added to the public folder.

ctrlplusb committed:
Merge branch 'master' into next

diondirza committed:
fix sw to exclude directory (#305)

ctrlplusb committed:
Merge branch 'master' into next

ctrlplusb committed:
Adds Dion Dirza as a contributor 🎉

ctrlplusb committed:
Fixes analyze.

ctrlplusb committed:
Docs and cleaning up.

ctrlplusb committed:
Updates docs and version.

ctrlplusb committed:
Removes code-split-component.

ctrlplusb committed:
Updates deps and fixes config refs.

ctrlplusb committed:
Huuuuuge refactor of project structure.  The 'src/' was always unnecessary.  Everything internal (tooling/docs) has been moved to an "internal" folder.  The rest lives at root.
The configuration has been given a massive once over too.  I really disliked how there were multiple ways the config values were being read throughout the project.  Everything has been changed now to use a config helper that lives in the '<projectroot>/config/' folder.  All things config lives in that folder now too.

ctrlplusb committed:
Fixes offline page.

ctrlplusb committed:
Cleans up and centralises util functions.

ctrlplusb committed:
Minor fixes.

ctrlplusb committed:
Renames config/get to config/getConfig
Removes CONF_ENV and refactors environment variable resolution to use NODE_ENV again.

tyler-dot-earth committed:
fixes typo: laoder -> loader (#353)

tyler-dot-earth committed:
fixes typo: server -> serve (#354)

ctrlplusb committed:
Huuuuuuuge update. Getting closer to v13.

datoml committed:
Fixes incorrect import path in Menu.test.js (#360)

diondirza committed:
Updates Error 404 component so that we can set 404 status on SSR renders (#357)

ctrlplusb committed:
Updates dependencies.

ctrlplusb committed:
Updates contributors and comments.

ctrlplusb committed:
Merge branch 'next' of https://github.com/ctrlplusb/react-universally into next

ctrlplusb committed:
Restructures and fixes tests.

ctrlplusb committed:
Cleaning up comments and docs.

Commit:
Fixes deprecated fallbackLoader (#365)

birkir committed:
Changes default host config to bind to all hosts (0.0.0.0) by default (#361)

sergiokopplin committed:
merge with next

diondirza committed:
fix typo flag (#369)

ctrlplusb commented:
Awesome man.  You are a legend for picking this up.  I'll review within the next few days. 💖

sergiokopplin committed:
removing flow from branch

sergiokopplin commented:
Hey @ctrlplusb, i couldn't find the problem to solve the bug.

Do you know if is related with react-jobs or with react-async-component?
I want to solve, but i need the first step, there is a lot of thing going there!

birkir committed:
Missing default polyfill.io features

datoml committed:
Updated react-router-dom to beta 7 (#381)

ctrlplusb committed:
Updates node version.
Adds check for none-support of HTML 5 browserhistory on the router.

ctrlplusb committed:
Merge branch 'next' of https://github.com/ctrlplusb/react-universally into next

ctrlplusb committed:
Synchronises the react-router-dom dep into yarn.lock

ctrlplusb committed:
Updates dependencies.

ctrlplusb committed:
Removes onlyIf and replaces with ifElse helper. fixes #362

ctrlplusb committed:
Update README.md

ctrlplusb committed:
Update README.md

birkir committed:
Fixes breaking context changes in RR4

birkir committed:
Rollback disableSSR. TODO: EnvVars this

ctrlplusb committed:
Changes polyfill.io into a features array specification.

diondirza committed:
Upgrade react-router-dom@4.0.0 and other deps (#396)

* update deps

* Using babel-env instead, still produce same result

* Fix not found implementation for stable react-router v4.0.0

* Restore and update deps

ctrlplusb committed:
Syncs yarn.lock with package.json

birkir commented:
I got react-jobs and react-async-components working in a mobx environment if you want to take a look: https://github.com/ueno-llc/starter-kit-universally/pull/1/files

ctrlplusb commented:
FYI, some BIG updates coming to both `react-async-component` and `react-jobs`.  I have a new interop layer that allows both of them to share the same tree walk parse, therefore only requiring a single parse. And it will resolve any number of nested async/job components. 🎉

birkir committed:
Fix Error404 test

sergiokopplin commented:
Sorry guys, vacation time.
Waiting the merge. o//

ctrlplusb committed:
Upgrades to latest react-async-component.
Reintroduces React Hot Loader 🎉
Adds milligram for some basic styling out of the box.
Adds a counter route for testing state based hot reloading.
Restructures route components.

ctrlplusb committed:
Adds 'es6' features to polyfill.io config.

ctrlplusb committed:
Removes aliasing of React libs and instead makes NODE_ENV get set to either 'development' or 'production' via the webpack config.
Introduces DEPLOYMENT environment variable to allow for the specification of target environment configuration files.

ctrlplusb committed:
Updates comments.

ctrlplusb committed:
Removes babel-preset-latest.

ctrlplusb committed:
staticContext on 404 route not required.

ctrlplusb commented:
Hey all, 

react-async-component upgraded.

Next up react-jobs - a big rewrite that will impact this PR in a big manner.

@sergiokopplin if you haven't already can you add me as being able to contribute to your PR. Thanks

sergiokopplin commented:
@ctrlplusb added. Thanks for the help!

ctrlplusb committed:
Refactors async components structure.

ctrlplusb committed:
Adds prettier 💄

ctrlplusb committed:
Updates nvm

ctrlplusb committed:
Adds prepush hook to run jest.

ctrlplusb committed:
Fixes prettier config to use prettier-eslint. Woops\!

ctrlplusb committed:
Removes unused eslint-flowtype plugin.

ctrlplusb committed:
Updates dependencies.

ctrlplusb committed:
Updates docs

ctrlplusb committed:
Moves all html page elements to DemoApp component. closes #390

ctrlplusb committed:
Removes 'host' from server start. closes #398

ctrlplusb committed:
Changes process.env build flags to be inline with expected string type of typical process.env member. closes #395

ctrlplusb committed:
Merge branch 'master' of https://github.com/ctrlplusb/react-universally

ctrlplusb committed:
Preps v13 release

ctrlplusb committed:
Update README.md

ctrlplusb committed:
Update README.md

ctrlplusb committed:
Updates dependencies.

ctrlplusb committed:
Merge branch 'master' of https://github.com/ctrlplusb/react-universally

ctrlplusb committed:
Update breaking changes

ctrlplusb committed:
Adds birkir to contributors list

ctrlplusb committed:
Adds @birkir to contributors on about us route.

ctrlplusb committed:
Changes over to all-contributors

ctrlplusb committed:
Orders contributors by name

sergiokopplin commented:
@ctrlplusb hey man, i'm back from vacations, i have some time to help here!

---

We're having some problems here.
The application must wait the request return to render the pages, but it's returning before the success. So, if you disable the JS in the browser, it will not work.

I still have the bug (might be a bug) in my application. Can you help with that?
I want to help in the merge, so i can upgrade my application here and also solve the bug.

Thanks again man. Awesome work here!

sergiokopplin commented on server/middleware/reactApplication/index.js:
```diff
@@ -0,0 +1,90 @@
+        helmet={Helmet.rewind()}
+        asyncComponents={{ state, STATE_IDENTIFIER }}
+        initialState={getState()}
+        jobsTate={{ state, STATE_IDENTIFIER }}
```
@Neophy7e hey man, i'm not sure about this details, i need some help from @ctrlplusb to understand this.

sergiokopplin commented:
@ctrlplusb i tried to merge the fork/redux-opinionated with the upstream/next but it has some conflicts that i'm not sure how to solve.

ctrlplusb commented:
No worries.  Good to have you back! :)

I'll make some time to do a big merge / update of that branch with the latest.  We will work against master now, as I put next into master - officially released v13.

I'll get back to you early next week. 👍

sergiokopplin commented:
Thanks man!

birkir commented:
I have it up and running, checkout the changes: https://github.com/ueno-llc/starter-kit-universally/commit/09af801006f21ad7e91a675ce8c4d44b02ea2ef7

sergiokopplin commented:
@birkir thanks man!

ChinW committed:
fix the "collectCoverageFrom" config of jest

birkir committed:
Merge pull request #418 from ChinW/master

fix the "collectCoverageFrom" config of jest

sergiokopplin committed:
[merge next] solving conflicts

sergiokopplin commented:
@ctrlplusb i've solved some conflicts on the repo with the next/merge in pair with @birkir.
I'll that a look on some errors that is coming in my terminal tomorrow!

sergiokopplin commented:
![image](https://cloud.githubusercontent.com/assets/2743180/25027296/e50dbaf4-2081-11e7-8c58-17fb1b3f7105.png)

sergiokopplin commented:
![image](https://cloud.githubusercontent.com/assets/2743180/25027297/ed28a118-2081-11e7-8e06-8e158aff7138.png)

sergiokopplin committed:
updating asyncPosts

birkir committed:
Prepare React 15.5 (#413)

* Update dependecies, change component prop-types.

* Helmet now has prop-types support.

* Update yarn lockfile

* Bumped react-router and forgot prop types in clientconfig.

* Bump enzyme and prop-types

* Bump react-async-component and enzyme

ctrlplusb commented:
@sergiokopplin sorry I haven't been able to give you time on this.  I have a client project I am pushing hard on at the moment.  Time will come.  Looks like you are doing a great job thusfar!

ctrlplusb committed:
Updates deps.

diondirza committed:
Another bug fixes, tweaks (#422)

* Fix lint

* code split naming strategy sample

* remove HTTP protocol to be more flexible

* Refactor polyfill.io
Due to counter route become broken in mobile device, use only default feature instead

* Fix lint while prettier revert changes

* update deps

birkir committed:
Add es6 back by default. (#429)

Commit:
Windows (#415)

* update eslintrc line-breaks for windows

* add prop-types as a dependency to prevent react-hot-loader from breaking build.
add cross-env for starting tasks which break on Windows.
add FAQ for when running develop and javascript files aren't loading even though the build is successful.
modify the clean task to execute a function rather than use the ${npm bin} exec command, which doesnst work on Windows.

* save exact package.json

* add cross-env to npm scripts

kdavh committed:
dev server host always 0.0.0.0, not host config

oyeanuj commented:
> Time will come

Hey @ctrlplusb! Just wanted to get a sense from you for when do you see this being merged and what parts need to happen before the merge? Looking forward to using it for a project of mine 😄

sergiokopplin commented:
@oyeanuj it is almost ready! I had some troubles to construct the async pages, but the configurations seems to be ready!

sergiokopplin committed:
merge with universally/master

sergiokopplin committed:
merge with universally/master - updating snapshots

sergiokopplin committed:
merge/next - adding async posts

Vincz commented on shared/components/DemoApp/AsyncPostsRoute/Post/Post.js:
```diff
@@ -69,12 +67,23 @@ function Post() {
+export default withJob({
+  work: (props) => {
+    const { id } = props.match.params;
+    return fetch(`http://jsonplaceholder.typicode.com/posts/${id}`).then(r => r.json());
```
The fetch function is only defined in the browser causing the job to failed on the server side. What I don't understand is why there is nothing in the server console saying the function doesn't exist.

sergiokopplin replied:
@Vincz yeah, you're right!! I'm missing something, but don't now how to solve.

strues replied:
Any luck with this? I just pulled down your branch to look into it

sergiokopplin replied:
Not yet! Not working properly! The redux functions are not plugged into the code. The server side rendering is not rendering 100%. Im working based on @birkir fork!

birkir replied:
Use axios for fetching data, its cancellable and also lightweight.

sergiokopplin committed:
merge/next - updating dependencies + updating async components + removing counter component

sergiokopplin committed:
merge/next - updating server side state

strues commented:
I was starting to dig in but had to drop my MacBook off at Apple for the second time in a month.  


On April 29, 2017 at 3:01:57 PM, Sérgio A. Kopplin (notifications@github.com(mailto:notifications@github.com)) wrote:

>  
>  
> @sergiokopplin commented on this pull request.
>  
>  
> In shared/components/DemoApp/AsyncPostsRoute/Post/Post.js(https://github.com/ctrlplusb/react-universally/pull/366#discussion_r114061711):
>  
> > // )(Post); -export default Post; +export default withJob({ + work: (props) => { + const { id } = props.match.params; + return fetch(`http://jsonplaceholder.typicode.com/posts/${id}`).then(r => r.json());  
>  
> Not yet! Not working properly! The redux functions are not plugged into the code. The server side rendering is not rendering 100%. Im working based on @birkir(https://github.com/birkir) fork!
>  
>  
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub(https://github.com/ctrlplusb/react-universally/pull/366#discussion_r114061711), or mute the thread(https://github.com/notifications/unsubscribe-auth/AF7kZfY6x7L3hzlC558OExG93-7JjNvIks5r06VFgaJpZM4MH3G-).
>  
>  
>  
>  
>

Vincz commented on shared/components/DemoApp/AsyncPostsRoute/Post/Post.js:
```diff
@@ -78,12 +78,14 @@ function Post({ jobResult }) {
   work: (props) => {
     const { id } = props.match.params;
-    return fetch(`http://jsonplaceholder.typicode.com/posts/${id}`).then(r => r.json());
+    return fetch(`http://jsonplaceholder.typicode.com/posts/${id}`)
```
The problem is still here. fetch function is undefined server side. You should use axios to perform the query as it works on both side. If you do, then the SSR works just fine.

Vincz commented:
Hi guys !
I have detected two problem:
- Still using "fetch" in the Post component. Fetch doesn't work server side. You should use Axios (working on both sides) to perform the json query.
- Client side, the redux store is populated from window.__APP_STATE__ which is great as we want the store to be prepopulated by the server. But, the server side counterparts is missing. In this file, we should get the store state https://github.com/sergiokopplin/react-universally/blob/feature/redux-opinionated/server/middleware/reactApplication/index.js then add it to the window in https://github.com/sergiokopplin/react-universally/blob/feature/redux-opinionated/server/middleware/reactApplication/ServerHTML.js

sergiokopplin commented:
Awesome work @Vincz. The axios thing is new for me, great catch! I'll update the rep! Thanks again

Vincz commented:
I'd like to send you the patch but I have no idea how to add commits to someone else PR (I'm a git newbie).

So, for the axios part, it would be:

    return axios(`http://jsonplaceholder.typicode.com/posts/${id}`)
      .then(r => r.data)
      .catch(err => console.error(err));

And for the redux state, 
in `server/middleware/reactApplication/index.js`

    <ServerHTML
        reactAppString={appString}
        nonce={nonce}
        helmet={Helmet.rewind()}
        storeState={store.getState()}
        jobsState={jobContext.getState()}
        asyncComponentsState={asyncContext.getState()}
    />

and in `server/middleware/reactApplication/ServerHTML.js`

line 43:

    const { storeState, jobsState, asyncComponentsState, helmet, nonce, reactAppString } = props;

line 64 (add):

    // Bind our redux store state so the client knows how to hydrate his one
    ifElse(storeState)(() => inlineScript(`window.__APP_STATE__=${serialize(storeState)};`)),

sergiokopplin commented:
@Vincz awesome!! thanks!
tomorrow i'll update the fork!

sergiokopplin committed:
merge/next - fixing server async functions

brianfeister commented:
Not sure if this is of any interest to you guys, but I grabbed the fork from this PR and added `material-ui` which was a bit difficult for me, being my second day on the job as a React developer. No idea if this aligns with your goals in maintaining branches with other ingredients in the recipe.

Went down a rabbit hole with the `material-ui` folks due to the complexity in the redux-opinionated fork:
https://github.com/nathanmarks/jss-theme-reactor/issues/41#issuecomment-298260293

https://github.com/brianfeister/react-universally/commit/2e80248a2c46b68fe38fa7c022dad33fd07a84de

sergiokopplin committed:
universally/merge - fixing async posts route

sergiokopplin commented:
hey @brianfeister, we should talk with @ctrlplusb first about your ideas. :)

sergiokopplin commented:
a73f171 fixed!
to test if it's working, disable the javascript on browser then open the post/route. The async functions are working properly fetching all the data on the server!

Commit:
chore: add redux, react-redux, redux-thunk to vendorDLL

ctrlplusb commented:
Sorry I have been a bit silent. Really warms my heart to see so many people pulling together on this one. Hope to help soon. 😘

strues commented:
@sergiokopplin submitted a PR under your fork with a small oversight i saw this AM.

sergiokopplin committed:
Merge pull request #1 from strues/feature/redux-opinionated

chore: add redux, react-redux, redux-thunk to vendorDLL

sergiokopplin commented:
Thanks @strues

diondirza committed:
Fix configuration for extracted style hash (#438)

threehams committed:
Fix broken link to Feature Branches docs. (#419)

diondirza committed:
Fix CORS host (#432)

birkir committed:
Merge pull request #433 from kdavh/public-dev-server-fix

dev server host always 0.0.0.0, not host config

Commit:
Removes reference to deprecated react PropTypes import and replaces with the prop-types package. Adds tests for Post route, post reducers and additional tests to home and about route.

Commit:
update snapshot

strues commented:
@sergiokopplin Another PR submitted on your branch. The old PropTypes from react were being used on the post component. I went ahead and added a few more tests as well.

sergiokopplin committed:
Merge pull request #2 from strues/feature/redux-opinionated

Feature/redux opinionated

sergiokopplin commented:
Thanks @strues!!!!! Awesome work!

sergiokopplin commented:
@ctrlplusb i'm not sure if is ready to merge, can you take a look? Thanks

strues commented:
It's certainly close to ready, if not completely ready. Havent noticed any other issues myself.

brianfeister committed:
Fix Router v2 Implementation

brianfeister commented:
I think I uncovered a minor bug where wrapping components differently than you're doing in the boilerplate causes the async `/post/:id` route to choke, I think this might be a better approach

https://github.com/sergiokopplin/react-universally/pull/3

sergiokopplin committed:
Merge pull request #3 from brianfeister/patch-1

Fix Router v2 Implementation

sergiokopplin commented:
Nice catch @brianfeister !!!

birkir committed:
Add color support for stdout (#443)

Passing colors from server to console in develop.

sergiokopplin committed:
Merge remote-tracking branch 'upstream/master' into feature/redux-opinionated

sergiokopplin commented:
Hey guys!
The branch has 1 problem which affects the render speed.

![image](https://cloud.githubusercontent.com/assets/2743180/26471683/108f70e2-4179-11e7-8433-1013e5241d7f.png)

I was looking as seems to be a problem with the async wrapper.
Can someone helps me with that? Not finding the problem.

sergiokopplin commented:
Fixed!

I changed this line from client/index:

```js
const rehydrateState = window.__JOBS_REHYDRATE_STATE__;
```

to:

```js
const rehydrateState = window.__JOBS_STATE__;
```

sergiokopplin committed:
master -> redux: fixing store problem

maryokhin commented on client/index.js:
```diff
@@ -30,7 +30,7 @@ const supportsHistory = 'pushState' in window.history;
 // Get any rehydrateState for the async components.
 // eslint-disable-next-line no-underscore-dangle
-const asyncComponentsRehydrateState = window.__ASYNC_COMPONENTS_REHYDRATE_STATE__;
+const asyncComponentsRehydrateState = window.__JOBS_STATE__;
```
Should this part in [server/middleware/reactApplication/ServerHTML.js](https://github.com/ctrlplusb/react-universally/blob/master/server/middleware/reactApplication/ServerHTML.js#L69) also reflect this?
```
 ifElse(asyncComponentsState)(() =>
      inlineScript(
        `window.__ASYNC_COMPONENTS_REHYDRATE_STATE__=${serialize(asyncComponentsState)};`,
      ),
    ),
```
also reflect this change?

maryokhin commented:
Running this branch as-is gives me a bunch of errors in the console, is that normal?

![screenshot 2017-05-31 13 48 06](https://cloud.githubusercontent.com/assets/4390231/26630649/eb8cf1be-4607-11e7-9fc9-7487412f4618.png)

sergiokopplin committed:
fixing stores

maryokhin commented:
@sergiokopplin yep, console is clean now.

maryokhin commented:
I tried writing my own stack of actions, reducers and components following the pattern in this branch...but completely failed 😞 maybe someone can explain why there is no `fetchPost` in my props? 
```javascript
const mapDispatchToProps = {
  fetch: fetchPlaylists,
};

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withJob({
    work: (props) => {
      console.log("work:", props);
      const { playlists, fetch } = props;
      console.log("fetch:", fetch);

      if (playlists) {
        return true;
      }

      return fetch();
    },
  }),
)(PlaylistList);
```
<img width="698" alt="screenshot 2017-06-01 00 05 39" src="https://cloud.githubusercontent.com/assets/4390231/26656158/43a97736-465e-11e7-864b-f641ad157503.png">

oyeanuj commented:
@sergiokopplin @ctrlplusb Ready to merge? 🎉

strues commented:
LGTM

ctrlplusb commented:
If @strues is happy lets go! 👍

ctrlplusb commented:
Well done to @sergiokopplin for all his hard work on this one, and a huge thanks to everyone else who helped along the way. 👍

sergiokopplin commented:
Awesome!!!! :)

sergiokopplin commented:
@ctrlplusb can i add my name on the contributors :p?

ctrlplusb commented:
Of course yes!  

Sorry I have been completely dropping the ball on managing this project at the moment.  I have so many other gears turning, with some exciting new developments.  I appreciate to the moon your contribution.  In fact, expect an invite. 👍

sergiokopplin commented:
Thanks @ctrlplusb!

sergiokopplin commented:
@ctrlplusb hey man! i've pushed a commit that was supposed to be a pr on the next branch to add me as contributor. sorry for that, my mistake! Can you add me on readme also? thanks

ctrlplusb commented:
No worries, my sincere apologies for not having you in there already!
